src/litserve/server.py

```diff
@@ -170,6 +170,8 @@ def run_streaming_loop(lit_api: LitAPI, lit_spec, request_queue: Queue, request_
 
         try:
             context = {}
+            if isinstance(lit_spec, OpenAISpec):
```

Comment:
we need to avoid making this specific to the OpenAISpec. The correct condition here is: if the spec implements populate_context, then we call it.